src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1729,6 +1729,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 return;
             }
 
+            var usage = ticket.GetUsage();
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
+            var audiences = ticket.GetAudiences();
+            if (string.Equals(usage, OpenIdConnectConstants.Usages.RefreshToken, StringComparison.Ordinal)) {
+                // For Refresh tokens
+                if ((clientNotification.IsSkipped &&
```

Comment:
> IMHO, it's much more obvious to log that the client was simply not authenticated than telling the developer the token was "potentially" issued to another client :smile:

I can understand wanting more detail about why the refresh token information wasn't sent to the caller, though in essence it's the same case - you don't know that the caller is the client that requested it... though I do see your point with wanting more verbose logging. I'll make that change.